js/store.js

```diff
@@ -349,21 +354,20 @@ const store = {
     return true;
   },
 
-  addFirstPartyLink(firstPartyHostname, thirdPartyHostname) {
+  async addFirstPartyLink(firstPartyHostname, thirdPartyHostname) {
     const firstParty = this.getWebsite(firstPartyHostname);
     if (!('thirdPartyHostnames' in firstParty)) {
       firstParty['thirdPartyHostnames'] = [];
     }
     if (!this.isFirstPartyLinkedToThirdParty(firstParty, thirdPartyHostname)) {
       firstParty['thirdPartyHostnames'].push(thirdPartyHostname);
-      this.setFirstParty(firstPartyHostname, firstParty);
+      await this.setFirstParty(firstPartyHostname, firstParty);
```

Comment:
I'm not sure what you mean here... What would be the array of Promises? You mean capturing `this.setFirstParty` into an array that persists across calls to `addFirstPartyLink`? Could you provide some pseudocode to help illustrate?